assemble/src/main/scripts/create-jshell.sh

```diff
@@ -0,0 +1,110 @@
+  export hadoop="accumulo/hadoop"
+    
+  # Does an auto-generated JShell config file exists?
+  if [[ ! -e "$jPath" ]]; then
+    echo "Cannot add APIs in $jPath"
+    echo "Please ensure jshell-init.jsh exists"
+    exit 1
+  fi
+  
+  # Is the source directory valid?
+  if [[ ! -d "$srcDir" ]]; then
+    echo "$srcDir is not a valid directory. Please make sure it exists."
+    rm "$jPath"
+    exit 1
+  fi
+  
+  # Does a valid JShell path and source directory exists?
+  if [[ -e "$jPath" ]] && [[ -d "$srcDir" ]]; then
+    # Add API category designator in jshell-init.jsh 
+    case "$srcDir" in
+        *"$client"*) echo "// Accumulo Client API" >> "$jPath";;
+        *"$data"*) echo "// Accumulo Data API" >> "$jPath";;
+        *"$security"*) echo "// Accumulo Security API" >> "$jPath";;
+        *"$mini"*) echo "// Accumulo Minicluster API" >> "$jPath";;
+        *"$hadoop"*) echo "// Accumulo Hadoop API" >> "$jPath";;
+        *) echo "// Other API" >> "$jPath";;
+    esac
+   
+    # Extract API info from provided source directory
+    mapfile -t api < <(find "$srcDir" -type f -name '*.java'| 
+                       xargs -n1 dirname| sort -u)
+   
+    # Load in API and format source directory into Java import statements
+    for apiPath in "${api[@]}"; do
+       printf "%s\n" "import ${apiPath##*/java/}.*" >> "$jPath"
+    done
+    sed -i '/^ *import / s#/#.#g' "$jPath"
+    echo " " >> "$jPath"
+  fi
+}
+
+function main(){
+  # Establish Accumulo's main base directory
+  SOURCE="${BASH_SOURCE[0]}"
+  while [[ -h "${SOURCE}" ]]; do
+    bin="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"
+    SOURCE="$(readlink "${SOURCE}")"
+    [[ "${SOURCE}" != /* ]] && SOURCE="${bin}/${SOURCE}"
+  done
+    
+  # Establish file and folder paths for JShell config
+  export scriptPath="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"; 
+  export mainBase=$( cd -P "${scriptPath}"/../../../.. && pwd );
+  export jPath="$mainBase/assemble/target/jshell-init.jsh"
+  export corePath="core/src/main/java/org/apache/accumulo/core"
+  export miniPath="minicluster/src/main/java/org/apache/accumulo"
+  export hadoopPath="hadoop-mapreduce/src/main/java/org/apache/accumulo"
+    
+  # Create path to Accumulo Public API Source Directories
+  export CLIENT="$mainBase/$corePath/client"
+  export DATA="$mainBase/$corePath/data"
+  export SECURITY="$mainBase/$corePath/security"
+  export MINI="$mainBase/$miniPath/minicluster"
+  export HADOOP="$mainBase/$hadoopPath/hadoop/mapreduce" 
+   
+  # Does an auto-generated JShell config file exists?
+  if [[ -e "$jPath" ]]; then
+     rm "$jPath"
+  fi
+    
+  # Create new jshell-init file and load in license header   
+  touch "$jPath"
+ 
+  # Create and add Accumulo APIs into API storage 
+  apiStorage=("$CLIENT" "$DATA" "$SECURITY" "$MINI" "$HADOOP")
+   
+  # Traverse through each source directory and load in Accumulo APIs
+  for srcDir in "${apiStorage[@]}"; do
+    addAccumuloAPI "$srcDir" "$jPath"
+  done
+  exit 0
```

Comment:
Normally at the end of a function, you'd want to let the return code of the last command propagate outside of the function. If this loop terminates abnormally, we shouldn't mask its return code with our own using `exit 0`, which forces it to terminate successfully, even if there was an error.

Also, I was thinking, you don't need to pass in the filename to the function or create it ahead of time. Instead, you can just create the file right here at the output of the whole for loop. That would simplify the implementation of the addAccumuloAPI command a lot.

```suggestion
    addAccumuloAPI "$srcDir"
  done > "$jPath"
```
One other suggestion I had was that you could move the printing of the section headers to outside that method, that way you don't need to create so many variables in that method to do the section header checks. You could either move the if statements to here, inside this for loop, before calling `addAccumuloAPI`, or you remove the loop, and do something like the following. You wouldn't even need all the extra variables this way, because you could just inline them:

```bash
  {
    echo "// section header here for client"
    addAccumuloAPI "$CLIENT"
    echo "// section header here for data"
    addAccumuloAPI "$DATA"
    # etc...
  } > "$jPath"
```